### ISSUE TITLE
Forgetting Devices using AWS Amplify in Android

### ISSUE THREAD

**Issue by leomuko** [aws-amplify/amplify-android#2682]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin

### Amplify Categories

Authentication

### Gradle script dependencies

<details>

```gradle dependencies
// Put output below this line
 implementation 'com.amplifyframework:core:2.14.6'
implementation 'com.amplifyframework:aws-auth-cognito:2.14.7'

```

</details>

### Environment information
Android Studio
Kotlin
Jetpack Compose

<details>

```
# Put output below this line


```

</details>

### Describe the bug

Summary:
Implementing a feature to forget devices after a user logs out is not behaving as expected when using AWS Amplify in Android. According to the Amplify documentation, the recommended method for achieving this functionality is to call `Auth.forgetDevice()` before signing out. However, while this successfully changes the status of the tracked device to "No," the device still continues to be displayed in the console, rather than being completely removed.

I tried to follow along with this https://docs.amplify.aws/android/build-a-backend/auth/remember-device/#terminology
but it did not yield the desired results.

Expected Behavior:
After calling `Auth.forgetDevice()` and signing out, the device should be completely removed from the AWS console, and its information should no longer be visible.

Actual Behavior:
The device status is correctly updated to "No," but the device continues to appear in the AWS console, indicating that it is not completely forgotten.

Workaround:
As of now, no known workaround exists to ensure that forgotten devices are completely removed from the AWS console. The issue persists even after following the recommended approach from the Amplify documentation.

### Reproduction steps (if applicable)

**Steps to Reproduce:**
1. Implement the forget device feature using `Auth.forgetDevice()` before calling the sign-out functionality.
2. Observe that the device status is correctly changed to "No" in the tracking system.
3. Check the AWS console to verify if the device is removed from the list after calling `Auth.forgetDevice()`.

### Code Snippet

```kotlin
// Put your code below this line.
Auth.forgetDevice()
```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

### Additional information and screenshots

**Additional Information:**
- AWS Amplify version: 2.14.7
- Android with Jetpack Compose

**Comment by tylerjroach** [comment]
Thank you for your report. Someone on our team will be taking a look.

**Comment by mattcreaser** [comment]
Hi @leomuko. I've confirmed that Amplify is clearing the flag but not properly forgetting the device. I'll work on making a fix here.

**Comment by mattcreaser** [comment]
This has been released in Amplify 2.14.10 🎉
